lib/application.js

```diff
@@ -54,9 +54,9 @@ var trustProxyDefaultSymbol = '@@symbol:trust_proxy_default';
 app.init = function init() {
   var router = null;
 
-  this.cache = {};
-  this.engines = {};
-  this.settings = {};
+  this.cache = Object.create(null);
+  this.engines = Object.create(null);
+  this.settings = Object.create(null);
```

Comment:
They are private, which is why they are not documented and the testing for them is tested though the public APIs that utilize them 👍

Reply:
Sounds good.

Out of curiosity, why aren't they prefixed with underscores to indicate that they aren't meant to be used publicly? Not related to this PR at all.

Reply:
Yes, it is curious 😀 They are before my time so cannot answer the why, lol. Even an underscore doesn't actually stop folks from using stuff, and a large project even ends up needing to be csreful modifying those things. Really only way to protect is an inside out obect, clever usage of closures, or the newer private class members. Idk if 5.x will change it, as it is late in the dev on it and probably need to scout round for the usages to make sure they are still provided in some way and truely private them in like 6.

I bet people are accessing settings directly, at least. Perhaps too cache if they wanted to clear it since there is no API to do so.

Reply:
Great answer. Thanks!